default/python/AI/MilitaryAI.py

```diff
@@ -183,7 +183,7 @@ def allocate(self, group, sys_id, min_rating, min_rating_vs_planets, take_any, m
         tup = (sys_id, min_rating, min_rating_vs_planets, take_any, max_rating)
         self.allocations.append(tup)
         self.allocation_by_groups.setdefault(group, []).append(tup)
-        self._remaining_rating -= min_rating
+        self._remaining_rating = rating_difference(self._remaining_rating,  min_rating)
```

Comment:
Maths changed here from ```x = x - y``` to what is basically ``` x = abs(x - y)```

Reply:
In this case, there is no check that remaining_rating > min_rating. You rely on external code / calling convention to not break this function which seems like a ticking time bomb.